Snippets/Core/Core_6/Routing/RoutingAPIs.cs

```diff
@@ -117,6 +118,15 @@ void DynamicEndpointMapping(EndpointConfiguration endpointConfiguration)
             #endregion
         }
 
+        void CustomDistributionStrategy(EndpointConfiguration endpointConfiguration)
+        {
+            #region Routing-CustomDistributionStrategy
+
+            var routing = endpointConfiguration.UnicastRouting();
+            routing.Mapping.SetMessageDistributionStrategy(new CustomStrategy(), messageType => messageType.GetInterfaces().Contains(typeof(IUseCustomDistributionStrategy)));
```

Comment:
this is the fast path? so do we need to add a note about caching the reflection?